src/database/schema/ChatSchema.ts

```diff
@@ -174,6 +175,28 @@ class ChatSchema implements BaseDbSchema {
     return Promise.resolve(this.fromDatabaseObject(rows.raw()[0]));
   }
 
+  static async getPendingMessages(db: SQLiteDatabase): Promise<ChatSchema[]> {
+    const selectQuery = `SELECT A.*,
+      B.username, 
+      B.country_code, 
+      B.created_at as user_created_at, 
+      B.updated_at as user_updated_at, 
+      B.last_active_at, 
+      B.profile_picture, 
+      B.bio, 
+      B.is_banned,
+      C.channel_type
+      FROM ${ChatSchema.getTableName()} A
+      INNER JOIN ${UserSchema.getTableName()} B
+      ON A.user_id = B.user_id
+      INNER JOIN ${ChannelListSchema.getTableName()} C
+      ON A.channel_id = C.id
+      WHERE A.status = 'pending';`;
+
+    const [{rows}] = await db.executeSql(selectQuery, ['pending Messages']);
+    return Promise.resolve(rows.raw().map(this.fromDatabaseObject));
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

A new static method `getPendingMessages` is introduced. It retrieves pending messages from the database by joining tables `ChatSchema`, `UserSchema`, and `ChannelListSchema`. The SQL query appears to be correct, but it's important to ensure that all fields are correctly named and exist in their respective tables. Also, make sure that the 'pending' status exists and is correctly handled in your application.
```diff
+   static async getPendingMessages(db: SQLiteDatabase): Promise<ChatSchema[]> {
+     const selectQuery = `SELECT A.*,
+       B.username, 
+       B.country_code, 
+       B.created_at as user_created_at, 
+       B.updated_at as user_updated_at, 
+       B.last_active_at, 
+       B.profile_picture, 
+       B.bio, 
+       B.is_banned,
+       C.channel_type
+       FROM ${ChatSchema.getTableName()} A
+       INNER JOIN ${UserSchema.getTableName()} B
+       ON A.user_id = B.user_id
+       INNER JOIN ${ChannelListSchema.getTableName()} C
+       ON A.channel_id = C.id
+       WHERE A.status = 'pending';`;
+ 
+     const [{rows}] = await db.executeSql(selectQuery, ['pending Messages']);
+     return Promise.resolve(rows.raw().map(this.fromDatabaseObject));
+   }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->